### PR TITLE
WIP - Add checkpoint_timeout warning

### PR DESCRIPTION
**Pull request by pgstef**
In the backup command, add a warning if `start-fast` is disabled and PG `checkpoint_timeout` is greater than `db-timeout`.
In such cases, we might timeout before the checkpoint occurs and the backup really starts.

Note: WIP regarding the tests part

### CODE REVIEW

**Comment by pgstef** [comment]
The initial test is failing:

```
2021-11-09 09:08:50.116 P00  ERROR: [125]: P1-T1/1 - vm=f33, module=db, test=db (err1-7.66s):
    
    STDOUT:
    run 1 - Db and dbProtocol()
        000.032s          L0117     create client
        000.038s          L0073     config load: /home/vagrant/test/bin/f33/pgbackrest backup:remote --lock-path=/home/vagrant/test/data-0/lock --log-path=/home/vagrant/test/data-0 --stanza=test1 --pg1-path=/home/vagrant/test/test-0/pg --pg1-database=testdb --remote-type=pg --process=0
        000.182s 000.144s L0097     create server
        000.201s 000.019s L0104     run process loop
    
    run 1/1 ------------- L0125 open and free database
        000.219s 000.187s L0127     create db
        000.224s 000.005s L0131     open db
    ==1552== Invalid read of size 1
    ==1552==    at 0x474FB0: pgClientQuery (client.c:274)
    ==1552==    by 0x40CAE5: dbQueryProtocol (protocol.c:88)
    ==1552==    by 0x46DEAC: protocolServerProcess (server.c:203)
    ==1552==    by 0x40E308: testRun (dbTest.c:104)
    ==1552==    by 0x422E75: main (test.c:219)
    ==1552==  Address 0x0 is not stack'd, malloc'd or (recently) free'd
    ==1552== 
    {
       <insert_a_suppression_name_here>
       Memcheck:Addr1
       fun:pgClientQuery
       fun:dbQueryProtocol
       fun:protocolServerProcess
       fun:testRun
       fun:main
    }
    ==1552== 
    ==1552== Exit program on first error (--exit-on-first-error=yes)
    
    TEST FAILED WITH TestError:
    
    --------------------------------------------------------------------------------
    EXPECTED VOID RESULT FROM STATEMENT: dbOpen(db)
    
    BUT GOT FileWriteError: unable to write to parent 0 write: [32] Broken pipe
    
    THROWN AT:
    /home/vagrant/test/repo/src/common/io/fdWrite.c:ioFdWrite:80:(trace log level required for parameters)
    /home/vagrant/test/repo/src/common/io/write.c:ioWriteFlush:(trace log level required for parameters)
    /home/vagrant/test/repo/src/protocol/client.c:protocolClientCommandPut:(trace log level required for parameters)
    /home/vagrant/test/repo/src/protocol/client.c:protocolClientFreeResource:(trace log level required for parameters)
    /home/vagrant/test/repo/test/src/module/db/dbTest.c:testRun:131:(void)
    test.c:main:(argListSize: 1, argList: (char *[]))
    --------------------------------------------------------------------------------
    
2021-11-09 09:08:50.531 P00   INFO: TESTS COMPLETED WITH 1 FAILURE(S) (21s)
```

I probably missed something in the macro (see commit f945098) but I can't find it out unfortunately :-(

**Comment by jreidthompson** [comment]
Hi Stefan,
it looks like #define HRNPQ_MACRO_VALIDATE_QUERY(sessionParam, versionParam, pgPathParam, archiveMode, archiveCommand) has a param for each of the subsequent query items, does it need one for checkpoint_timeout also?

**Comment by pgstef** [comment]
> Hi Stefan, it looks like #define HRNPQ_MACRO_VALIDATE_QUERY(sessionParam, versionParam, pgPathParam, archiveMode, archiveCommand) has a param for each of the subsequent query items, does it need one for checkpoint_timeout also?

Hi Reid,
Thanks for the suggestion.

I don't think it would fix the issue, I'm using the default value for the 1st test:
`{.session = sessionParam, .function = HRNPQ_GETVALUE, .param = "[0,4]", .resultInt = 300},`

File: test/src/common/harnessPq.h [hunk 1]
```diff
@@ -68,24 +68,27 @@ Macros for defining groups of functions that implement various queries and comma
...
+    {.session = sessionParam, .function = HRNPQ_FTYPE, .param = "[4]", .resultInt = HRNPQ_TYPE_INT},                               \
     {.session = sessionParam, .function = HRNPQ_GETVALUE, .param = "[0,0]", .resultZ = STRINGIFY(versionParam)},                   \
     {.session = sessionParam, .function = HRNPQ_GETVALUE, .param = "[0,1]", .resultZ = pgPathParam},                               \
     {.session = sessionParam, .function = HRNPQ_GETVALUE, .param = "[0,2]", .resultZ = archiveMode == NULL ? "on"                  \
         : archiveMode},                                                                                                            \
     {.session = sessionParam, .function = HRNPQ_GETVALUE, .param = "[0,3]", .resultZ = archiveCommand == NULL ? PROJECT_BIN        \
         : archiveCommand},                                                                                                         \
+    {.session = sessionParam, .function = HRNPQ_GETVALUE, .param = "[0,4]", .resultInt = 300},                                     \
```
Comment (dwsteele) [inline test/src/common/harnessPq.h:91]: This needs to be `.resultZ = STRINGIFY(300)`. See an example at line 85.

Postgres passes all values back as zero-terminated strings so we must do the same.

Comment (pgstef) [reply test/src/common/harnessPq.h]: Oh ok. Thanks for the explanation, it indeed solved the problem :-)

**Comment by dwsteele** [comment]
Reopened as #1564.

**Pull request closed without merging**